cmd/attested-get/main.go

```diff
@@ -0,0 +1,157 @@
+// and print the verified measurements and the response payload.
+//
+// Currently only works for Azure TDX but is straight-forward to expand.
+//
+// Usage:
+//
+//   go run cmd/attested-get/main.go --addr=https://instance_ip:port
+//
+// Can also save the verified measurements and the response body to files:
+//
+//   go run cmd/attested-get/main.go --addr=https://instance_ip:port --out-measurements=measurements.json --out-response=response.txt
+//
+
+import (
+	"encoding/asn1"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/flashbots/cvm-reverse-proxy/common"
+	"github.com/flashbots/cvm-reverse-proxy/internal/atls"
+	"github.com/flashbots/cvm-reverse-proxy/internal/attestation/variant"
+	"github.com/flashbots/cvm-reverse-proxy/proxy"
+	"github.com/urfave/cli/v2" // imports as package "cli"
+)
+
+var flags []cli.Flag = []cli.Flag{
+	&cli.StringFlag{
+		Name:  "addr",
+		Value: "https://localhost:7936",
+		Usage: "TEE server address",
+	},
+	&cli.StringFlag{
+		Name:  "out-measurements",
+		Value: "",
+		Usage: "Output file for the measurements",
+	},
+	&cli.StringFlag{
+		Name:  "out-response",
+		Value: "",
+		Usage: "Output file for the response payload",
+	},
+	&cli.BoolFlag{
+		Name:  "log-debug",
+		Value: false,
+		Usage: "log debug messages",
+	},
+}
+
+func main() {
+	app := &cli.App{
+		Name:   "attested-get",
+		Usage:  "Get verified measurements",
+		Flags:  flags,
+		Action: runClient,
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func runClient(cCtx *cli.Context) (err error) {
+	logDebug := cCtx.Bool("log-debug")
+	addr := cCtx.String("addr")
+	outMeasurements := cCtx.String("out-measurements")
+	outResponse := cCtx.String("out-response")
+
+	// Setup logging
+	log := common.SetupLogger(&common.LoggingOpts{
+		Debug:   logDebug,
+		JSON:    false,
+		Service: "attested-get",
+		Version: common.Version,
+	})
+
+	if !strings.HasPrefix(addr, "https://") {
+		return errors.New("address needs to start with https://")
+	}
+
+	log.Info("Getting verified measurements from " + addr + " ...")
+
+	// Prepare aTLS stuff
+	issuer, err := proxy.CreateAttestationIssuer(log, proxy.AttestationAzureTDX)
+	if err != nil {
+		log.Error("could not create attestation issuer", "err", err)
+		return err
+	}
+
+	tlsConfig, err := atls.CreateAttestationClientTLSConfig(issuer, []atls.Validator{})
+	if err != nil {
+		log.Error("could not create atls config", "err", err)
+		return err
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	client := &http.Client{Transport: tr}
+	resp, err := client.Get(addr)
+	if err != nil {
+		return err
+	}
+	certs := resp.TLS.PeerCertificates
+
+	// Extract the aTLS variant and measurements from the TLS connection
+	atlsVariant, extractedMeasurements, err := proxy.GetMeasurementsFromTLS(certs, []asn1.ObjectIdentifier{variant.AzureTDX{}.OID()})
```

Comment:
I guess the only improvement here would be to add the usual `--attestation-type`
I'd add a TODO